packages/client/src/PhantomClient.ts

```diff
@@ -212,39 +240,65 @@ export class PhantomClient {
       if (!this.config.organizationId) {
         throw new Error("organizationId is required to sign a transaction");
       }
-      // Transaction is always a string (encoded via parsers)
-      const encodedTransaction = transactionParam;
-
-      // Check if this is an EVM transaction using the network ID
-      const isEvmTransaction = isEthereumChain(networkIdParam);
-
-      let submissionConfig: SubmissionConfig | null = null;
 
-      if (includeSubmissionConfig) {
-        submissionConfig = deriveSubmissionConfig(networkIdParam) || null;
+      // SubmissionConfig is used to: 1) submit the transaction onchain, 2) derive spending limits
+      const submissionConfig = deriveSubmissionConfig(networkIdParam);
 
-        // If we don't have a submission config, the transaction will only be signed, not submitted
-        if (!submissionConfig) {
-          console.error(
-            `No submission config available for network ${networkIdParam}. Transaction will be signed but not submitted.`,
-          );
-        }
+      if (!submissionConfig) {
+        throw new Error(`SubmissionConfig could not be derived for network ID: ${networkIdParam}`);
       }
 
-      // Get network configuration with custom derivation index
       const networkConfig = getNetworkConfig(networkIdParam, derivationIndex);
 
       if (!networkConfig) {
         throw new Error(`Unsupported network ID: ${networkIdParam}`);
       }
 
+      // Transaction is always a string (encoded via parsers)
+      const encodedTransaction = transactionParam;
```

Comment:
Nit: we are doubly reassigning with `encodedTransaction` and `preparedTransaction`. We should just keep the latter.